src/block-management/block-manager.js

```diff
@@ -128,6 +132,12 @@ export class BlockManager extends React.Component<PropsType, StateType> {
 		SafeArea.addEventListener( 'safeAreaInsetsForRootViewDidChange', this.onSafeAreaInsetsUpdate );
 	}
 
+	componentDidUpdate( prevProps: PropsType ) {
+		if ( this.props.blockCount > prevProps.blockCount ) {
```

Comment:
I'm curios about this one.
As I understand, this code fixes the issue of typing on a long paragraph block. How is the block count involved on that? 🤔

Reply:
you are %100 right to be curious about it. this is not related with appending to a long paragraph, it is related with the other issue about caret staying outside of viewable are when a split occurs. we are detecting if there's a new block added. I just added a safety belt here although this is not necessary %95 of the time, actually I tried now and couldn't reproduce the issue, so maybe it looks like we don't need this immediately. I'll remove it, we can add later if there's an upcoming issue about it, that way the intent will be more obvious.